### PR TITLE
Add IDL patch for WebCodecs

### PR DESCRIPTION
**Pull request by tidoust**
Reverts the IDL to what it was before the introduction of dictionary attribute.

### CODE REVIEW

File: ed/idlpatches/webcodecs.idl.patch [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+diff --git a/ed/idl/webcodecs.idl b/ed/idl/webcodecs.idl
+index dbb3cc2de..16d6bd5ed 100644
+--- a/ed/idl/webcodecs.idl
++++ b/ed/idl/webcodecs.idl
+@@ -161,7 +161,6 @@ dictionary VideoEncoderConfig {
+   [EnforceRange] unsigned long displayHeight;
+   HardwareAcceleration hardwareAcceleration = "allow";
+   DOMString scalabilityMode;
+-  BitrateMode bitrateMode = "variable";
```
Comment (dontcallmedom) [inline ed/idlpatches/webcodecs.idl.patch:19]: should we only revert the bits that are problematic? do we have a policy or precedents?

Comment (tidoust) [reply ed/idlpatches/webcodecs.idl.patch]: I think I confused the previous commit for webcodecs.idl in webref and the previous commit for the source spec. There have been more than one commit that changes the IDL in the WebCodecs spec between two updates in webref in practice.

Anyway, I rolled back the rollback. The attribute is now preserved.

File: ed/idlpatches/webcodecs.idl.patch [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+diff --git a/ed/idl/webcodecs.idl b/ed/idl/webcodecs.idl
+index dbb3cc2de..4fc4f975f 100644
+--- a/ed/idl/webcodecs.idl
++++ b/ed/idl/webcodecs.idl
+@@ -275,23 +275,22 @@ enum AudioSampleFormat {
+ [Exposed=(Window,DedicatedWorker)]
+ interface VideoFrame {
+   constructor(CanvasImageSource image, optional VideoFrameInit init = {});
+-  constructor(sequence<PlaneInit> planes,
```
Comment (dontcallmedom) [inline ed/idlpatches/webcodecs.idl.patch:19]: my previous comment applies also to this and other bits below - do we revert only the problematic bits (dictionaries used as attributes) or all the changes that were brought in at the same time?

Comment (tidoust) [reply ed/idlpatches/webcodecs.idl.patch:19]: My worry is that the other updates are more intertwined and were done at the same time. I don't know whether the IDL makes a lot of sense if you only apply part of the changes. I would rather err on the safe side and freeze the IDL to what it was before.

Comment (dontcallmedom) [reply ed/idlpatches/webcodecs.idl.patch:19]: Thanks - that makes sense to me

**Review by dontcallmedom — APPROVED** [review]

**Pull request merged**